drivers/ioexpander/gpio.c

```diff
@@ -596,6 +596,64 @@ int gpio_pin_register(FAR struct gpio_dev_s *dev, int minor)
   return register_driver(devname, &g_gpio_drvrops, 0666, dev);
 }
 
+/****************************************************************************
+ * Name: gpio_pin_register_byname
```

Comment:
please modify gpio_pin_register to call gpio_pin_register_byname internally to avoid the code duplication.

Reply:
Done